scripts/scanResults.js

```diff
@@ -0,0 +1,193 @@
+    );
+    process.stdout.write(`\rDownloading block: ${blockNum}`);
+  }
+  console.log();
+  fs.writeFileSync(`./${votesFile}`, JSON.stringify(txs, null, 2));
+  return txs;
+};
+
+const getTxData = () => {
+  if (fs.existsSync(`./${votesFile}`)) {
+    return require(`../${votesFile}`);
+  }
+  return downloadTxs();
+};
+
+const slimProposal = ({ 
+  title, proposalId, boothAddress, yesBoxAddress, noBoxAddress
+}) => ({
+  title, proposalId, boothAddress, yesBoxAddress, noBoxAddress
+});
+
+const downloadProposals = async () => {
+  return new Promise((resolve, reject) => 
+    https.get(proposalData, (res) => {
+      let raw = '';
+      res.on('data', d => raw += d);
+      res.on('end', () => resolve(JSON.parse(raw)));
+      res.on('error', e => reject(e));
+    })
+  );  
+};
+
+const getProposals = async () => {
+  if (fs.existsSync(`./${proposalsFile}`)) {
+    return require(`../${proposalsFile}`);
+  }
+  
+  const rawProps = await downloadProposals();
+  const proposals = rawProps.contents.proposals
+    .filter(p => p.proposalId)
+    .map(slimProposal);
+
+  fs.writeFileSync(`./${proposalsFile}`, JSON.stringify(proposals, null, 2));
+  return proposals;
+};
+
+const getVotes = (tx) => {
+  // vote is a 4th argument (index 3) to castBallot/withdraw call
+  const votes = ethers.utils.defaultAbiCoder.decode(
+    booth.functions.castBallot.inputs.map(i => i.type),
+    bufferToHex(tx.inputs[0].msgData.slice(4)) // cut a func sig
+  )[3].div(factor18).toNumber();
+
+  return isWithdraw(tx) ? -votes : votes;
+};
+
+const getProposalByBox = (proposals, boxAddress) =>
+  proposals.find((prop) => 
+      prop.yesBoxAddress === boxAddress 
+      || prop.noBoxAddress === boxAddress
+  );
+
+const countByNumberOfVotes = (arr) => 
+  arr.reduce((r, v) => {
+    r[v[1]] = (r[v[1]] || 0) + 1;
+    return r;
+  }, {});
+
+const getBoxAddress = (tx, voter) => {
+  if (isWithdraw(tx)) {
+    return bufferToHex(ripemd160(tx.inputs[0].script));
+  }
+  return tx.outputs.find(o => o.address !== voter && o.color === 4).address;
+};
+
+const getProposalId = (tx, proposals, voter) => {
+  const boxAddr = getBoxAddress(tx, voter);
+  const { proposalId } = getProposalByBox(proposals, boxAddr) || {};
+  if (proposalId) {
+    return proposalId;
+  }
+  console.warn(
+    'Unknown proposal vote',
+    JSON.stringify({ boxAddr })
+  );
+};
+
+(async () => {
+  const txs = await getTxData();
+  const proposals = await getProposals();
+  const distr = {};
+  const voters = new Set();
+  txs.forEach(t => {
+    const tx = Tx.fromRaw(t.raw);
+    const withdrawalTx = isWithdraw(tx);
+    if (isVote(tx) || withdrawalTx) {
+      const voter = tx.inputs[1].signer; // balance card signer
+      voters.add(voter);
+      const proposalId = getProposalId(tx, proposals, voter);
+      if (!proposalId) {
+        return;
+      }
+      
+      const proposalVotes = distr[proposalId] = distr[proposalId] || {};
```

Comment:
(personal preference) To me is a bit too much technical debt.
I see, from right to left evaluation, that this will assign the same value to `distr[proposalId]` and `proposalVotes`.
But I'd rather keep it on two lines, just to clarify the intent.